fili-core/src/test/java/com/yahoo/bard/webservice/models/druid/client/impl/TestDruidWebService.java

```diff
@@ -151,6 +153,8 @@ public void postDruidQuery(
         } catch (IOException e) {
             failure.invoke(e);
         }
+
+        return ConcurrentUtils.constantFuture(null);
```

Comment:
I think this `return` should be lifted up and happen inside the `catch` (2 lines up) wrapping the `IOException`, as well as happening on each of the branches inside that `try` block as well.